### PR TITLE
Update JsonCPP to 1.8.3

### PR DESCRIPTION
**Pull request by nerzhul**
Update embedded jsoncpp to 1.8.3 and fix our code as we are using long time deprecated, but not marked as is, methods

### CODE REVIEW

File: lib/jsoncpp/json/UPDATING [hunk 1]
```diff
@@ -1,16 +1,7 @@
 #!/bin/sh
 cd ..
-svn co https://jsoncpp.svn.sourceforge.net/svnroot/jsoncpp/trunk/jsoncpp jsoncpp
-svn up jsoncpp
+git clone https://github.com/open-source-parsers/jsoncpp -b 1.8.3
```
Comment (SmallJoker) [inline lib/jsoncpp/json/UPDATING]: We don't need the entire history of this library, do we?

Comment (nerzhul) [reply lib/jsoncpp/json/UPDATING]: i don't know i just updated this file, what option do you want me to add ?

File: src/mods.cpp [hunk 2]
```diff
@@ -393,9 +394,12 @@ bool ModMetadata::load(const std::string &root_path)
...
-	Json::Reader reader;
 	Json::Value root;
-	if (!reader.parse(is, root)) {
+	Json::CharReaderBuilder builder;
+	builder.settings_["collectComments"] = false;
+	std::string errs;
+
+	if (!Json::parseFromStream(builder, is, &root, &errs)) {
 		errorstream << "ModMetadata[" << m_mod_name << "]: failed read data "
 			"(Json decoding failure)." << std::endl;
```
Comment (SmallJoker) [inline src/mods.cpp]: Also write `errs` to the logger, as we've got information about the cause why it failed.

File: lib/jsoncpp/json/UPDATING (hunk 1 above)
Comment (SmallJoker) [reply lib/jsoncpp/json/UPDATING]: `--depth 1`, but that's a minor request, as this update script will be only executed once per year, if even.

**Pull request merged**